### PR TITLE
Set the logout context access token for sso logout context

### PR DESCRIPTION
**Pull request by RAVEENSR**
## Purpose
This PR fixes the https://github.com/wso2/analytics-apim/issues/872 issue.

## Goals
Fix bugs.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java [hunk 1]
```diff
@@ -249,6 +262,14 @@ public Response loginAppNamePost(String appName
             errorDTO.setError(IdPClientConstants.Error.INTERNAL_SERVER_ERROR);
             errorDTO.setDescription("Error in login to the uri '" + appName + "'. Error: " + e.getMessage());
             return Response.serverError().entity(errorDTO).build();
+        } catch (ConfigurationException e) {
+            LOG.debug("Error occurred while reading configs from deployment.yaml. " + e.getMessage()) ;
+            ErrorDTO errorDTO = new ErrorDTO();
+            errorDTO.setError(IdPClientConstants.Error.INTERNAL_SERVER_ERROR);
+            errorDTO.setDescription("Error occurred while reading configs from deployment.yaml. " + e.getMessage());
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity(errorDTO)
```
Comment (AnuGayan) [inline components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java]: shall we came this one line??

Comment (RAVEENSR) [reply components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java]: Fixed in commit 683d433.

File: components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java [hunk 2]
```diff
@@ -249,6 +262,12 @@ public Response loginAppNamePost(String appName
             errorDTO.setError(IdPClientConstants.Error.INTERNAL_SERVER_ERROR);
             errorDTO.setDescription("Error in login to the uri '" + appName + "'. Error: " + e.getMessage());
             return Response.serverError().entity(errorDTO).build();
+        } catch (ConfigurationException e) {
+            LOG.debug("Error occurred while reading configs from deployment.yaml. " + e.getMessage()) ;
```
Comment (TanyaM) [inline components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java:266]: Need to check isDebugEnabled

Comment (RAVEENSR) [reply components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java:266]: Fixed in a0108b09b0816017989673d72ee76c8f34cce7f5

File: components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java [hunk 3]
```diff
@@ -177,10 +177,23 @@ public Response loginAppNamePost(String appName
...
                                     -1);
-                    NewCookie logoutContextAccessToken = AuthUtil
-                            .cookieBuilder(AuthRESTAPIConstants.WSO2_SP_TOKEN, accessTokenSecondHalf,
-                                    AuthRESTAPIConstants.LOGOUT_CONTEXT + appContext, true, true,
-                                    -1);
+
+                    NewCookie logoutContextAccessToken;
+                    IdPClientConfiguration authConfigurations = DataHolder.getInstance()
+                            .getConfigProvider()
+                            .getConfigurationObject(IdPClientConfiguration.class);
```
Comment (1akshitha) [inline components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java:184]: You might be able to reformat this without breaking the max line length

File: components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java [hunk 4]
```diff
@@ -177,10 +177,23 @@ public Response loginAppNamePost(String appName
...
+                    if (authConfigurations.isSsoEnabled()) {
+                        logoutContextAccessToken = AuthUtil
+                                .cookieBuilder(AuthRESTAPIConstants.WSO2_SP_TOKEN, accessTokenSecondHalf,
+                                        AuthRESTAPIConstants.LOGOUT_CONTEXT + AuthRESTAPIConstants.LOGOUT_SSO_CONTEXT +
+                                                appContext , true, true, -1);
+                    } else {
+                        logoutContextAccessToken = AuthUtil
+                                .cookieBuilder(AuthRESTAPIConstants.WSO2_SP_TOKEN, accessTokenSecondHalf,
+                                        AuthRESTAPIConstants.LOGOUT_CONTEXT + appContext, true, true,
+                                        -1);
```
Comment (1akshitha) [inline components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java:194]: Shall we reformat this to the previous line?

File: components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java (hunk 3 above)
Comment (RAVEENSR) [reply components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java:184]: Fixed in 6ed65129793669c8b287520f081c75f00fd0e92f

File: components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java (hunk 4 above)
Comment (RAVEENSR) [reply components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java:194]: Fixed in 6ed65129793669c8b287520f081c75f00fd0e92f

**Pull request merged**